packages/components/src/tab-panel/index.tsx

```diff
@@ -100,28 +102,12 @@ export function TabPanel( {
 	) => {
 		child.click();
 	};
-	const selectedTab = tabs.find( ( { name } ) => name === selected );
+	const selectedTab = getTab( tabName ) || firstEnabledTab;
 	const selectedId = `${ instanceId }-${ selectedTab?.name ?? 'none' }`;
 
-	useEffect( () => {
-		const firstEnabledTab = tabs.find( ( tab ) => ! tab.disabled );
-		const initialTab = tabs.find( ( tab ) => tab.name === initialTabName );
-		if ( ! selectedTab?.name && firstEnabledTab ) {
-			handleTabSelection(
-				initialTab && ! initialTab.disabled
-					? initialTab.name
-					: firstEnabledTab.name
-			);
-		} else if ( selectedTab?.disabled && firstEnabledTab ) {
-			handleTabSelection( firstEnabledTab.name );
-		}
-	}, [
-		tabs,
-		selectedTab?.name,
-		selectedTab?.disabled,
-		initialTabName,
-		handleTabSelection,
-	] );
+	// useEffect( () => {
+	// 	setTabName( selectedTab?.name );
+	// }, [ selectedTab?.name ] );
```

Comment:
Without this hook, `onSelect` is not called on non user interactions such as changes to prop `tabName`, or removal of selected tab from `tabs`. Having this hook creates a side effect of calling the `onSelect` method twice for user actions such as click.

Please suggest ideas to handle this.